platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/IdpHandler.java

```diff
@@ -182,15 +257,136 @@ public Object getAttribute(String name) {
         return handlerResult;
     }
 
+    private boolean isEcpEnabled(ServletRequest request) {
+        String acceptHeader = ((HttpServletRequest) request).getHeader(HttpHeaders.ACCEPT);
+        String paosHeader = ((HttpServletRequest) request).getHeader(PAOS);
+        return acceptHeader != null && paosHeader != null && acceptHeader.contains(PAOS_MIME)
+                && paosHeader.contains(PAOS_NS) && paosHeader.contains(ECP_NS);
+    }
+
+    private HandlerResult doPaosRequest(ServletRequest request, ServletResponse response) {
+        HttpServletResponse httpServletResponse = (HttpServletResponse) response;
+        HandlerResult handlerResult = new HandlerResult(HandlerResult.Status.REDIRECTED, null);
+        handlerResult.setSource("idp-" + "-" + SOURCE);
+        String paosHeader = ((HttpServletRequest) request).getHeader(PAOS);
+
+        boolean wantChannelBind = paosHeader.contains(
```

Comment:
Yeah I left them there for future additions so that I'd know what wasn't implemented. I'll add a comment.